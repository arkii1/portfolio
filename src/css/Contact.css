.contact {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 1rem;
    padding: 0 2rem;
}

.contact__header {
    font-size: 2.2rem;
}

.contact__caption {
    text-align: center;
}

.contact__form {
    display: flex;
    flex-direction: column;
    gap: 1rem;
}

@media screen and (max-width:999px) {
    .contact__form {
        width: calc(100% - 2rem);
    }
}

@media screen and (min-width:1000px) {
    .contact__form {
        width: 40rem;
    }
}

.contact__form__message {
    height: 8rem;
}

.contact__form__submit {
    align-self: end;
    font-size: 0.75rem;
    width: 100px;
    padding: 0.25rem;
    background-color: var(--c-black);
    color: var(--c-white);
    border: 2px solid var(--c-white);
    border-radius: 5px;
}

.contact__form__submit:hover {
    background-color: var(--c-white);
    color: var(--c-black);
    border: 2px solid var(--c-white);
    cursor: pointer;
}