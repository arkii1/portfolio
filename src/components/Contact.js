import React from "react"
import "../css/Contact.css"

function Contact() {
  return (
    <div className="contact">
      <h2 className="contact__header">Contact</h2>
      <hr className="break" />
      <p className="contact__caption">
        If you have an oppurtunity to share, or would like or discuss anything,
        you can leave your details below.
      </p>
      <form className="contact__form">
        <input
          className="contact__form__name"
          type="text"
          id="name"
          name="name"
          placeholder="Name"
        />
        <input
          className="contact__form__email"
          type="text"
          id="email"
          name="email"
          placeholder="E-mail"
        />
        <textarea
          className="contact__form__message"
          id="message"
          name="message"
          placeholder="Message"
        />
        <input
          className="contact__form__submit"
          type="submit"
          id="submit"
          name="submit"
        />
      </form>
    </div>
  )
}

export default Contact
