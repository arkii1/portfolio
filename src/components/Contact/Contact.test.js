import React from "react"
import ReactDOM from "react-dom/client"
import { cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import Contact from "./Contact"

afterEach(cleanup)

it("renders without crashing", () => {
  const container = document.createElement("div")
  const root = ReactDOM.createRoot(container)
  root.render(<Contact />)
})

it("matches snapshot", () => {
  const tree = renderer.create(<Contact />).toJSON()
  expect(tree).toMatchSnapshot()
})
