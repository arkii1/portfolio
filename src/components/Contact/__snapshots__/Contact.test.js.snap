// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="contact"
>
  <h2
    className="contact__header"
  >
    Contact
  </h2>
  <hr
    className="break"
  />
  <p
    className="contact__caption"
  >
    If you have an oppurtunity to share, or would like or discuss anything, you can leave your details below.
  </p>
  <form
    className="contact__form"
  >
    <input
      className="contact__form__name"
      id="name"
      name="name"
      placeholder="Name"
      type="text"
    />
    <input
      className="contact__form__email"
      id="email"
      name="email"
      placeholder="E-mail"
      type="text"
    />
    <textarea
      className="contact__form__message"
      id="message"
      name="message"
      placeholder="Message"
    />
    <input
      className="contact__form__submit"
      id="submit"
      name="submit"
      type="submit"
    />
  </form>
</div>
`;
